tapeagents/tools/browser.py

```diff
@@ -375,10 +378,26 @@ def _save_last_screenshot(self, image) -> str:
         return img_path
 
     def reset(self):
-        self._env.step("goto('about:blank')")
+        try:
+            self._env.step("goto('about:blank')")
+        except TargetClosedError as e:
+            logger.warning(f"Browser page/context closed during reset: {e}")
+            # Browser is closed, nothing to reset
+            pass
 
     def run_browser_action(self, action_text: str) -> PageObservation:
-        obs_dict, reward, terminated, truncated, info = self._env.step(action_text)
+        try:
```

Comment:
Is it a good way to handle errors in a sense of traces generation? I mean that this way we will have some amount of traces that could contains correct and useful actions but still end up with failure like that and zero reward. Maybe it could be better to discard these traces at all by catching such errors in the orchestrator loop and skipping them?

Reply:
You're right, let's leave it that way